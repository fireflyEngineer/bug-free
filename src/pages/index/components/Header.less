.Header {
  transform: translate(0, 0);
  transition: all 1s;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  box-sizing: border-box;

  &-Top {
    padding: 10px 36px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &.isHide {
    transform: translate(0, -100%);
  }

  &-Logo {
    height: 40px;
  }

  &-Divider {
    height: 40px;
    width: 100%;
    background: rgb(100, 109, 203);
  }

  &-GetStarted {
    padding: 6PX;
    border: 1px solid rgb(100, 109, 203);
    background: none;
    color: rgb(100, 109, 203);
    border-radius: 10PX;
  }

  &-Icons {
    i {
      cursor: pointer;
      font-size: 20PX;
      margin: 0 2px;
      color: #666;
    }
  }
}