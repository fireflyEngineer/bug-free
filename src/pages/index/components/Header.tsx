import withHeaderContainer from './HeaderContainer'
import './Header.less'

export interface Props {
  isSaved: boolean,
  isSignIn: boolean,
  visible?: boolean
}

export const Header: React.FC<Props> = ({
  isSaved,
  isSignIn,
  visible = true,
}) => (
  <section className={`Header ${visible ? '' : 'isHide'}`}>
    <div className="Header-Top">
      <img src="/image/logo.png" alt="logo" className="Header-Logo" />
      <div className="Header-Icons">
        {isSignIn ? <>
          <i className="iconfont icon-search" />
          <i className={`iconfont ${isSaved ? 'icon-bookmark-save' : 'icon-bookmark'}`} />  
        </> : <button className="Header-GetStarted">Get Started</button>}
      </div>
    </div>
    <div className="Header-Divider" />
  </section>
)

export default withHeaderContainer(Header)