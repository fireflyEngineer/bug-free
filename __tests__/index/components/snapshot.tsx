import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../../../src/pages/index/components/Header'


it('renders homepage unchanged', () => {
  const tree = renderer.create(<Header />).toJSON()
  expect(tree).toMatchSnapshot()
})