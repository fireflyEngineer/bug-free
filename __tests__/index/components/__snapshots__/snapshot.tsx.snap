// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div
  className="Header"
>
  <img
    alt="logo"
    className="Header-Logo"
    src="/image/logo.png"
  />
  <div
    className="Header-Icons"
  >
    <button>
      Get Started.
    </button>
  </div>
</div>
`;
